00_Base/src/trigger/AsyncReceiverApi.ts

```diff
@@ -9,28 +9,22 @@ export class AsyncReceiverApi extends BaseClientApi {
   async postAsyncResponse(
     url: string,
     body: any,
-    params?: OcpiParams,
+    params: OcpiParams,
   ): Promise<OcpiResponse<void> | null> {
-    if (params) {
-      params.authorization = await this.getAuthToken(
-        params.fromCountryCode,
-        params.fromPartyId,
-        params.toCountryCode,
-        params.toPartyId,
-      );
-      this.baseUrl = url;
-      return this.create(
-        OcpiEmptyResponse,
-        {
-          async: true,
-          additionalHeaders: this.getOcpiHeaders(params),
-        },
-        body,
-      );
-    } else {
-      return this.createRaw<OcpiResponse<void>>(url, body, {}).then(
-        (response) => response.result,
-      );
-    }
+    params.authorization = await this.getAuthToken(
```

Comment:
I almost feel like we should have some sort of request interceptor that auto fills in the auth header given the ocpi params.

For the other push triggers (sessions, cdrs, locations, etc.) do they look up the auth token themselves similar to this?

Reply:
@langiole It is a very good point. I am also struggling on this question. 

Currently cdrs, sessions and locations always broadcast. You can find the related part in each broadcaster, e.g., [SessionBroadcaster](https://github.com/citrineos/citrineos-ocpi/blob/main/00_Base/src/broadcaster/session.broadcaster.ts) and it calls this method [broadcastToClients](https://github.com/citrineos/citrineos-ocpi/blob/b249e20de3ba9af8cdd57d1e12ec8c0c503d1e7c/00_Base/src/trigger/BaseClientApi.ts#L259) to get tokens and endpoints. So get token does not happens in any client api scripts.

Currently AsyncReceiverApi and PutChargingProfile are the only 2 places which send to a single target client and do this getAuthToken. 

I was thinking about create sender scripts for aync response and charging profile which are similar to broadcasters then we move getAuthToken into senders?

Reply:
I think what is here is sufficient.